### ISSUE TITLE
BLYM: Cannot start game

### ISSUE THREAD

**Issue by n0samu** [ruffle-rs/ruffle#10180]
### Describe the bug

This game is only playable in BlueMaxima's Flashpoint, or in [the online version of Flashpoint here](https://ooooooooo.ooo/static/?e7c9bd7c-a6d8-b5f2-d6b5-01500034d9b0).

After clicking the Play button in BLYM, the game shows nothing but a black screen.

After opening Flashpoint, you can use this command to run the game in Ruffle:
`ruffle --proxy=http://localhost:22500/ http://www.gameshot.org/BLYM/BLYM0913_640x480.swf`

Here is a log file from running the game: [blym.log](https://github.com/ruffle-rs/ruffle/files/11005412/blym.log)

### Expected behavior

The game should display the menu.

### Affected platform

Desktop app

### Operating system

Windows 10

### Additional information

The problem seems related either to type coercion or the unimplemented DefineFont4 (TLF text) tag.

**Comment by n0samu** [comment]
[You can also test this game in the online version of Flashpoint now](https://ooooooooo.ooo/static/?e7c9bd7c-a6d8-b5f2-d6b5-01500034d9b0).

As of nightly 2023-07-25, the new error was:
```
ERROR run_frame:run_all_phases_avm2: ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "enterFrame", class: flash.events::Event, ptr: 0x20b05e28730 }) to handler FunctionObject(FunctionObject { ptr: 0x20b794f8320 }) : ReferenceError: Error #1065: Variable flash.text.engine::TextBlock is not defined.
        at flashx.textLayout.elements::Configuration$cinit()
        at global$init() [TU=frame1]
        at flashx.textLayout.elements::TextFlow$cinit()
        at global$init() [TU=frame1]
        at flashx.textLayout.conversion::PlainTextImporter/importFromString()
        at flashx.textLayout.conversion::PlainTextImporter/importToFlow()
        at flashx.textLayout.conversion::TextConverter$/importToFlow()
        at com.spilgames.api.components.TextFields::AutoFitTextFieldEx/updateProperties()
        at com.spilgames.api.components.TextFields::AutoFitTextFieldEx/validateNow()
        at com.spilgames.api.components.TextFields::AutoFitTextFieldEx()
        at com.spilgames.api.components.TextFields::LocalizedTextFieldEx()
        at com.spilgames.api.components.impl::BrandLanguageSelectImpl/setupDropDown()
        at com.spilgames.api.components.impl::BrandLanguageSelectImpl/set direction()
        at com.spilgames.bs.comps::LanguageSelect/set direction()
        at Main$/addLanguageSelect()
        at titleMC/ini()
        at Main/mcode()
```
This probably would've been fixed by #12326. But there was a new regression in nightly 2023-07-26: #12366

**Comment by n0samu** [comment]
After #11971, the errors are:
```
ERROR run_frame:run_all_phases_avm2: ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "localeChanged", class: flash.events::DataEvent, ptr: 0x20dfe6b22e0 }) to handler FunctionObject(FunctionObject { ptr: 0x20dfe4f5d10 }) : ArgumentError: Error #1063: Argument count mismatch on flash.events::DataEvent(). Expected 1, got 5.
        at flash.events::DataEvent/clone()
        at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
        at com.spilgames.api.connection::Connection/onLocaleChanged()
        at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
        at com.spilgames.api.components::LocaleManager/initialise()
        at com.spilgames.api.components::BrandSystem/addLanguageXML()
        at com.spilgames.api.components::BrandSystem/registerGameLocalizationPack()
        at com.spilgames.api.components::BrandSystem/setGameLocalizationPack()
        at com.spilgames.api.components::BrandSystem/onBrandLocalizationPackLoaded()
ERROR run_frame:run_all_phases_avm2: ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "localeChanged", class: flash.events::DataEvent, ptr: 0x20dfec90710 }) to handler FunctionObject(FunctionObject { ptr: 0x20dfe4f5d10 }) : ArgumentError: Error #1063: Argument count mismatch on flash.events::DataEvent(). Expected 1, got 5.
        at flash.events::DataEvent/clone()
        at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
        at com.spilgames.api.connection::Connection/onLocaleChanged()
        at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
        at com.spilgames.api.components::LocaleManager/updateElements()
        at com.spilgames.api.components::BrandSystem/checkComplete()
        at com.spilgames.api.components::BrandSystem/onConnectionReady()
        at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
        at com.spilgames.api.connection::Connection/http://www.spilgames.com/games/flash/api/internal::servicesLoadedAndConfiguredCallback()
        at com.spilgames.api.connection.services::ServiceManager/configureRegisteredServices()
        at com.spilgames.api.connection.services::ServiceManager/servicesLoadedCallback()
        at com.spilgames.api.connection.services::ServiceLoader/onUnpackedServiceDone()
        at com.spilgames.api.connection.services::ServiceLoader/loadUnpackedServices()
        at com.spilgames.api.connection.services::ServiceLoader/onServicePackReady()
        at flash.events::EventDispatcher/flash::events::EventDispatcher::dispatchEvent()
        at com.spilgames.api.services.impl::ServicePack/onServiceLoadInit()
ERROR run_frame:run_all_phases_avm2: ruffle_core::avm2::events: Error dispatching event EventObject(EventObject { type: "enterFrame", class: flash.events::Event, ptr: 0x20d8d6965f0 }) to handler FunctionObject(FunctionObject { ptr: 0x20d8adc2da0 }) : ReferenceError: Error #1069: Property replaceText not found on flash.text.engine.TextElement and there is no default value.
        at flashx.textLayout.elements::SpanElement/http://ns.adobe.com/textLayout/internal/2008::createContentElement()
        at flashx.textLayout.elements::ParagraphElement/http://ns.adobe.com/textLayout/internal/2008::createTextBlock()
        at flashx.textLayout.elements::ParagraphElement/http://ns.adobe.com/textLayout/internal/2008::getTextBlock()
        at flashx.textLayout.compose::ComposeState/createTextLine()
        at flashx.textLayout.compose::ComposeState/composeNextLine()
        at flashx.textLayout.compose::BaseCompose/composeParagraphElementIntoLines()
        at flashx.textLayout.compose::ComposeState/composeParagraphElement()
        at flashx.textLayout.compose::BaseCompose/composeBlockElement()
        at flashx.textLayout.compose::BaseCompose/composeInternal()
        at flashx.textLayout.compose::ComposeState/composeInternal()
        at flashx.textLayout.compose::BaseCompose/composeTextFlow()
        at flashx.textLayout.compose::ComposeState/composeTextFlow()
        at flashx.textLayout.compose::StandardFlowComposer/http://ns.adobe.com/textLayout/internal/2008::callTheComposer()
        at flashx.textLayout.compose::StandardFlowComposer/internalCompose()
        at flashx.textLayout.compose::StandardFlowComposer/updateToController()
        at flashx.textLayout.compose::StandardFlowComposer/updateAllControllers()
        at com.spilgames.api.components.TextFields::AutoFitTextFieldEx/updateProperties()
        at com.spilgames.api.components.TextFields::AutoFitTextFieldEx/validateNow()
        at com.spilgames.api.components.TextFields::AutoFitTextFieldEx()
        at com.spilgames.api.components.TextFields::LocalizedTextFieldEx()
        at com.spilgames.api.components.impl::BrandLanguageSelectImpl/setupDropDown()
        at com.spilgames.api.components.impl::BrandLanguageSelectImpl/set direction()
        at com.spilgames.bs.comps::LanguageSelect/set direction()
        at Main$/addLanguageSelect()
        at titleMC/ini()
        at Main/mcode()
```

**Comment by n0samu** [comment]
Fixed by #13948
